packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -33,7 +34,7 @@ export enum Tracing {
   DISABLED = "Disabled"
 }
 
-export interface FunctionProps {
+export interface FunctionProps extends EventInvokeConfigOptions {
```

Comment:
This would mix destinations with retry/event age which are different concepts. The fact that it exists in another CF resource (because different versions/aliases can have different configurations) should be hidden from the user. Even in the Lambda console those are separated.

Not sure a user would expect to find destinations under `invokeConfig`, a pure CF term.

I would also argue that a flat props interface helps discoverability in the IDE, this was once a guideline here: https://github.com/aws/aws-cdk/blob/561cffbf9a37f0522c132e24d160ecec21e8d44d/docs/src/writing-constructs.rst#implementation-details

Let me know what you think is best here and I will adapt.

Reply:
All good points. Let's leave these splatted.